spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -437,20 +480,33 @@ private RabbitTemplate buildRabbitTemplate(RabbitProducerProperties properties)
 	 * @param baseQueueName   The base name for the queue (including the binder prefix, if any).
 	 * @param routingKey  The routing key for the queue.
 	 * @param autoBindDlq true if the DLQ should be bound.
+	 * @param properties the properties.
 	 */
-	private void autoBindDLQ(final String baseQueueName, String routingKey, String prefix, boolean autoBindDlq) {
+	private void autoBindDLQ(final String baseQueueName, String routingKey, RabbitCommonProperties properties) {
+		boolean autoBindDlq = properties.isAutoBindDlq();
 		if (this.logger.isDebugEnabled()) {
 			this.logger.debug("autoBindDLQ=" + autoBindDlq
 					+ " for: " + baseQueueName);
 		}
 		if (autoBindDlq) {
-			String dlqName = constructDLQName(baseQueueName);
-			Queue dlq = new Queue(dlqName);
+			String dlqName;
+			if (properties.getDeadLetterQueueName() == null) {
+				dlqName = constructDLQName(baseQueueName);
+			}
+			else {
+				dlqName = properties.getDeadLetterQueueName();
+			}
+			Queue dlq = new Queue(dlqName, true, false, false, queueArgs(dlqName, properties, true));
 			declareQueue(dlqName, dlq);
-			final String dlxName = deadLetterExchangeName(prefix);
+			String dlxName = deadLetterExchangeName(properties);
 			final DirectExchange dlx = new DirectExchange(dlxName);
 			declareExchange(dlxName, dlx);
-			declareBinding(dlqName, BindingBuilder.bind(dlq).to(dlx).with(routingKey));
+			if (properties.getDeadLetterRoutingKey() == null) {
+				declareBinding(dlqName, BindingBuilder.bind(dlq).to(dlx).with(routingKey));
+			}
+			else {
+				declareBinding(dlqName, BindingBuilder.bind(dlq).to(dlx).with(properties.getDeadLetterRoutingKey()));
```

Comment:
Looks like duplicate code and `BindingBuilder` can be extracted to the variable and we would have only once `declareBinding()` call.
But, yeah, minor and just IMO...